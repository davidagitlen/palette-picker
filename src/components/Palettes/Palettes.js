import React from "react";
import "./Palettes.css";
import Palette from "../Palette/Palette";

const Palettes = ({palettes, editPalette, trashPalette, selectPalette}) => {
  const paletteList = palettes.map(palette => 
    <Palette 
      key={palette.id}
      palette={palette}
      editPalette={editPalette}
      trashPalette={trashPalette}
      selectPalette={selectPalette}
    />)
  return (
    <article className="Palettes">
      <h3>Palettes: Click to Edit</h3>
      {paletteList}
    </article>
  );
};

export default Palettes;