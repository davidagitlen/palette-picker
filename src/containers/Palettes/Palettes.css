.Palettes > h3 {
  background-color: #7383C0;
  height: 50px;
}

.Palettes {
  align-content: flex-end;
  flex-direction: row;
  border: 1px solid black;
  height: max-content;
  margin: 50px 10px 25px auto;
  width: 40%;
  border: 1px solid black;
}