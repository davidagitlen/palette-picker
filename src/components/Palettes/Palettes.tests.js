import React from 'react';
import ReactDOM from 'react-dom';
import Palettes from './Palettes';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});