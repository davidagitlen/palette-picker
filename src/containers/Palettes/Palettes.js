import React from "react";
import "./Palettes.css";
import Palette from "../Palette/Palette";

const Palettes = () => {
  return (
    <article className="Palettes">
      <h3>Palettes</h3>
      <Palette />
    </article>
  );
};

export default Palettes;