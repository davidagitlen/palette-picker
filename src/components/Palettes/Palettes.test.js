import React from 'react';
import { shallow } from 'enzyme';
import Palettes from './Palettes';

describe('Palettes', () => {

  let wrapper, palettes;

  beforeEach(() => {
    palettes = [
      {
      hex_1: "#3192EC",
      hex_2: "#77B8AB",
      hex_3: "#910E31",
      hex_4: "#D8C86A",
      hex_5: "#6E16C2",
      id: 34,
      palette: "one",
      project_id: 1,
      project_name: "Blah blah"
      },
      {
      hex_1: "#3192EC",
      hex_2: "#77B8AB",
      hex_3: "#910E31",
      hex_4: "#D8C86A",
      hex_5: "#6E16C2",
      id: 34,
      palette: "two",
      project_id: 2,
      project_name: "Blib blob"
      }
    ];
    wrapper = shallow(
      <Palettes 
        palettes={palettes}
        editPalette={jest.fn()}
        trashPalette={jest.fn()}
        selectPalette={jest.fn()}
      />
      );
  }); 

  it('should match the snapshot with data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });  
});