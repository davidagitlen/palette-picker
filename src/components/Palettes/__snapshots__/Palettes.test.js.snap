// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes should match the snapshot with data passed in correctly 1`] = `
<article
  className="Palettes"
>
  <h3>
    Palettes: Click to Edit
  </h3>
  <Palette
    editPalette={[MockFunction]}
    key="34"
    palette={
      Object {
        "hex_1": "#3192EC",
        "hex_2": "#77B8AB",
        "hex_3": "#910E31",
        "hex_4": "#D8C86A",
        "hex_5": "#6E16C2",
        "id": 34,
        "palette": "one",
        "project_id": 1,
        "project_name": "Blah blah",
      }
    }
    selectPalette={[MockFunction]}
    trashPalette={[MockFunction]}
  />
  <Palette
    editPalette={[MockFunction]}
    key="34"
    palette={
      Object {
        "hex_1": "#3192EC",
        "hex_2": "#77B8AB",
        "hex_3": "#910E31",
        "hex_4": "#D8C86A",
        "hex_5": "#6E16C2",
        "id": 34,
        "palette": "two",
        "project_id": 2,
        "project_name": "Blib blob",
      }
    }
    selectPalette={[MockFunction]}
    trashPalette={[MockFunction]}
  />
</article>
`;
